test/integration/SessionCacheServiceSpec.scala

```diff
@@ -157,7 +158,12 @@ class SessionCacheSpec extends IntegrationTestsSpec with MockitoSugar with Mongo
 
       val cache = await(sessionCache.cacheRepo.findById(request))
 
-      val expectedJson                   = toJson(CachedData(registerWithEoriAndIdResponse = Some(rd)))
+      val expectedJson = toJson(
+        CachedData(
+          registerWithEoriAndIdResponse = Some(rd),
+          submissionCompleteDetails = Some(SubmissionCompleteData(None, Some(rd.responseCommon.processingDate)))
+        )
+      )
```

Comment:
Can we also get a test for when submission complete data is already populated with a SubscriptionDetails to prove that our code retains the existing data and only overwrites the processing date?